content/blog/2023/2024-01-27-deserialize-with-jsoncreator.md

```diff
@@ -0,0 +1,465 @@
+    public ResponseEntity<Page<UserData>> getPagedUser(
+                    @RequestParam(defaultValue = "0") int page,
+                    @RequestParam(defaultValue = "20") int size) {
+        List<UserData> usersList = userService.getUsers();
+
+        // First let's split the List depending on the pagesize
+        int totalCount = usersList.size();
+        int startIndex = page * size;
+        int endIndex = Math.min(startIndex + size, totalCount);
+
+        List<UserData> pageContent = usersList.subList(startIndex, endIndex);
+
+        Page<UserData> employeeDtos = 
+                new PageImpl<>(pageContent, PageRequest.of(page, size), totalCount);
+
+        return ResponseEntity.ok()
+                .body(employeeDtos);
+    }
+````
+Here, the `Page` class is an interface in the `org.springframework.data.domain` package.
+When we make a GET request to the endpoint `http://localhost:8083/data/userdetails/page`, we see the JSON response as below:
+````json
+{
+    "content": [
+        {
+            "id": 1000,
+            "firstName": "Abel",
+            "lastName": "Doe",
+            "city": "NYC",
+            "createdDate": "2024-01-26"
+        },
+        {
+            "id": 1001,
+            "firstName": "Abuela",
+            "lastName": "Marc",
+            "city": "NYC",
+            "createdDate": "2024-01-26"
+        },
+        {
+            "id": 1002,
+            "firstName": "Adi",
+            "lastName": "Nath",
+            "city": "Cape Town",
+            "createdDate": "2024-01-26"
+        },
+        {
+            "id": 1003,
+            "firstName": "Asia",
+            "lastName": "Hines",
+            "city": "Cape Town",
+            "createdDate": "2024-01-26"
+        },
+        {
+            "id": 1004,
+            "firstName": "Frida",
+            "lastName": "Stone",
+            "city": "Cape Town",
+            "createdDate": "2024-01-26"
+        }
+        // More elements here
+    ],
+    "pageable": {
+        "sort": {
+            "empty": true,
+            "sorted": false,
+            "unsorted": true
+        },
+        "offset": 0,
+        "pageNumber": 0,
+        "pageSize": 20,
+        "paged": true,
+        "unpaged": false
+    },
+    "last": false,
+    "totalElements": 45,
+    "totalPages": 3,
+    "size": 20,
+    "number": 0,
+    "sort": {
+        "empty": true,
+        "sorted": false,
+        "unsorted": true
+    },
+    "first": true,
+    "numberOfElements": 20,
+    "empty": false
+}
+````
+The endpoint returned us some pagination metadata like the `totalElements`, `totalPages`, `sort`, `size`.
+**Here, we see that the application has a total of 45 records, which is divided into 3 pages where each page has a maximum of 20.
+This JSON response gives us the first 20 elements from the List.**
+
+
+Next, let's write a Spring Boot test to understand how the deserialization happens when we call the GET endpoint using `TestRestTemplate`.
```

Comment:
This is a little confusing. Why do we need deserialization when calling a GET endpoint. Reading further, it turns out that we only need the annotation to write the tests. The GET endpoint itself does not need the annotation. Let's explain that.